open_diloco/train_fsdp.py

```diff
@@ -181,10 +181,16 @@ def tokenize_function(data):
     )
 
 
-def get_model(config: Config) -> LlamaForCausalLM:
+def get_model(config: Config) -> GPT:
     # Load model
-    config_model = LlamaConfig.from_pretrained(config.path_model, attn_implementation=config.attn_implementation)
-    return LlamaForCausalLM.from_pretrained(pretrained_model_name_or_path=config.path_model, config=config_model)
+    if isinstance(config.llama_config, ModelConfig):
+        llama_config = config.llama_config
+    else:
+        with open(config.llama_config) as f:
+            llama_config = ModelConfig(**json.load(f))
+
+    llama_config.attention_impl = config.attention_impl
+    return GPT(llama_config)
```

Comment:
Hrmm would the initialisation be the same across diloco workers?

Reply:
yes indeed, is there a way to use hf hub to push the init ckpt like we did before ?

Reply:
seems like GPT inherits from nn.Module instead of transformers PreTrainedModel so it doesnt have the `from_pretrained` function. Maybe we can change it to allow loading from hub